homework-2/if/script.js

```diff
@@ -0,0 +1,124 @@
+//
+// +'10' + 1
+// // Будет 11. Потому как мы привели строку к числу с помощью знака "+".
+//
+// undefined + 2
+// // Консоль выдаст NaN, потому что при преобразовании в число undefined становится NaN.
+//
+// null + 5
+// // Консоль выдаст 5, потому что при преобразовании в число null становится 0.
+//
+// true + undefined
+// // Консоль выдаст NaN, потому что при преобразовании в число undefined становится NaN.
+
+
+
+// 1. Если переменная равна “hidden”, присвоить ей значение “visible”, иначе - “hidden”.
+var someVar = "hidden";
+
+if (someVar == "hidden") {
+  someVar = "visible";
+} else {
+  someVar = "hidden";
+}
+
+alert(someVar);
+
+// 2. Используя if, записать условие:
+//   a. если переменная равна нулю, присвоить ей 1;
+//   b. если меньше нуля - строку “less then zero”;
+//   c. если больше нуля - используя оператор “присвоение”, переменную умножить на 10 (использовать краткую запись).
+
+let someLet = 5;
+
+if (someLet == 0) {
+  alert(someLet = 1);
+} else if (someLet < 0) {
+  alert("less than zero");
+} else if (someLet > 0) {
+  alert(someLet = someLet * 10);
+}
+
+// 3. Дан объект let car = { name: 'Lexus', age: 10, create: 2008, needRepair: false }.
+// Написать условие если возраст машины больше 5 лет то нужно вывести в консоль сообщение 'Need Repair' и свойство needRepair в объекте car изменить на true; иначе изменить на false.
+
+let car = {
+  name: 'Lexus',
+  age: 10,
+  create: 2008,
+  needRepair: false
+}
+
+if (car.age > 5) {
+  console.log("Need Repair");
+  car.needRepair = true;
+} else {
+  console.log("No need to Repair");
+  car.needRepair = false;
+}
+
+// 4. Дан объект let item = { name: 'Intel core i7', price: '100$', discount: '15%' }.
+// Написать условие если у item есть поле discount и там есть значение то в объекте item создать поле priceWithDiscount и записать туда цену с учетом скидки и вывести ее в консоль, обратите внимание что поля discount и price это строки и вам из них нужно получить числа чтобы выполнить расчет. иначе если поля discount нет то вывести просто поле price в консоль.
+
+let item = {
+  name: 'Intel core i7',
+  price: '100$',
+  discount: "15%"
+}
+
+if (item.discount != undefined || null || "") {
+  item.priceWithDiscount = parseInt(item.price) - (parseInt(item.price) / 100 * parseInt(item.discount));
+  console.log(item.priceWithDiscount);
+} else if(item.discount) {
+  console.log(item.price);
+}
+
+// 5. Дан следующий код:
+// let product = { name: “Яблоко”, price: “10$” };
+// let min = 10; // минимальная цена
+// let max = 20; // максимальная цена
+// Написать условие если цена товара больше или равна минимальной цене и меньше или равна максимальной цене то вывести в консоль название этого товара, иначе вывести в консоль что товаров не найдено.
+
+let product = {
+  name: "Яблоко",
+  price: "10$"
+}
+
+let min = 10;
+let max = 20;
+
+if (parseInt(product.price) >= min && parseInt(product.price) <= max) {
```

Comment:
would be simpler if you'd use `let productPrice = parseInt(product.price); if(productPrice >= min) .....` In this case you'd have done operation parseInt once not twice